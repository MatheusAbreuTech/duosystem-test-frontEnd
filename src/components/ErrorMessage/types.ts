import { FieldErrors } from 'react-hook-form'

export type ErrorMessageProps = {
  errors: FieldErrors
  field: string
}
