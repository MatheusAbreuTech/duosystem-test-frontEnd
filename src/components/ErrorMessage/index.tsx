import { FieldErrors } from 'react-hook-form'
import * as S from './styles'

interface ErrorMessageProps {
  errors: FieldErrors
  field: string
}

const ErrorMessage = ({ errors, field }: ErrorMessageProps) => {
  if (!errors[field]) {
    return null
  }
  return <S.Error>{errors[field]?.message?.toString()}</S.Error>
}

export default ErrorMessage
