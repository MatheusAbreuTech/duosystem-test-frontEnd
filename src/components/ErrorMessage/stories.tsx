import { Meta, StoryObj } from '@storybook/react'
import ErrorMessage from '.'

export default {
  title: 'ErrorMessage',
  component: ErrorMessage,
  args: {
    field: 'testField',
    errors: {
      testField: {
        message: 'mensagem de erro'
      }
    }
  }
} as Meta

export const Default: StoryObj = {}
