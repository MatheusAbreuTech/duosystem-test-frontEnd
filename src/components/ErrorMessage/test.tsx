import { render } from '@testing-library/react'
import { FieldErrors } from 'react-hook-form'

import ErrorMessage from '.'

describe('<ErrorMessage />', () => {
  it('should display an error message when there is an error in the field', () => {
    const errors: FieldErrors = {
      campoTeste: {
        type: 'manual',
        message: 'teste'
      }
    }

    const { getByText, queryByText, container } = render(
      <ErrorMessage errors={errors} field="campoTeste" />
    )

    const errorMessage = getByText('teste')
    expect(errorMessage).toBeInTheDocument()

    const errorMessage2 = queryByText('testeeee')
    expect(errorMessage2).toBeNull

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should not display an error message when there is not an error', () => {
    const noErrors = {}

    const { queryByText } = render(
      <ErrorMessage errors={noErrors} field="campoTeste" />
    )

    const errorMessage = queryByText('Este Ã© um erro de teste')
    expect(errorMessage).toBeNull()
  })
})
