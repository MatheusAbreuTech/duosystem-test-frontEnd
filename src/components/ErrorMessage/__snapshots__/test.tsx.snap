// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should display an error message when there is an error in the field 1`] = `
.c0 {
  color: red;
  font-size: 1.4rem;
}

<span
  class="c0"
>
  teste
</span>
`;
